book.md

```diff
@@ -2110,16 +2112,101 @@ architectural state. [Could we find a good reference that explains
 micro-architectural versus architectural state in more detail? Is "Computer
 Architecture: A Quantitative Approach" the best reference available?]{.todo}
 
+**Faulting instructions**\index{faulting instructions} are instructions that
```

Comment:
I like this description!

Purely for completeness (and this is just me thinking out loud, so feel free to disregard), I wonder if it makes sense to mention _wrong-path events (WPEs)/instructions_ here, cf. "Understanding The Effects of Wrong-Path Memory
References on Processor Performance" https://people.inf.ethz.ch/omutlu/pub/mutlu_wmpi04.pdf (Sec. 4.2 Usefulness of Wrong-path References and Sec. 5 Related Work), "Wrong Path Events and Their Application to Early Misprediction Detection and Recovery" (slides, quick to skim: https://people.inf.ethz.ch/omutlu/pub/mutlu_micro04_talk.pdf). This may be a personal/subjective to me, but I recall immediately thinking of WPEs and and eerie similarity ("hey, we used to use that for performance!").